src/core/ucc_context.c

```diff
@@ -793,20 +778,22 @@ ucc_status_t ucc_context_create_proc_info(ucc_lib_h                   lib,
         }
     }
 
-    for (i = 0; i < ctx->n_tl_ctx; i++) {
+    n_tl_ctx = ctx->n_tl_ctx;
+    for (i = 0; i < n_tl_ctx; i++) {
         tl_ctx = ctx->tl_ctx[i];
         tl_lib = ucc_derived_of(tl_ctx->super.lib, ucc_tl_lib_t);
         if (tl_lib->iface->context.create_epilog) {
             status = tl_lib->iface->context.create_epilog(&tl_ctx->super);
-            if (UCC_OK != status) {
+            if (UCC_OK == status) {
+                created_ctx_counter++;
+            } else {
                 if (ucc_tl_is_required(lib, tl_lib->iface, 1)) {
                     ucc_error("ctx create epilog for %s failed: %s",
                               tl_lib->iface->super.name, ucc_status_string(status));
                     goto error_ctx_create;
                 } else {
                     ucc_debug("ctx create epilog for %s failed: %s",
                               tl_lib->iface->super.name, ucc_status_string(status));
-                    tl_lib->iface->context.destroy(&tl_ctx->super);
```

Comment:
why we don't destroy context in this case?

Reply:
In my opinion we shouldn't call ctx->destroy at this point because the ctx is not fully setup (because this happens before ctx create epilog).
The implementation of ctx create epilog itself should take care of the cleanup in case of an error that happens inside it.

Reply:
Actually we might want/need here and elsewhere (see [comment](https://github.com/openucx/ucc/pull/818#issuecomment-1669372142)) a function that cleans up only whats been allocated by ctx init (and not init epilog)...

Reply:
ok, please add short comment to context create epilog description that TL is responsible to cleanup resources if context create epilog fails

Reply:
I put back the `destroy` at this line
see https://github.com/openucx/ucc/pull/818#issuecomment-1695843289